src/skillsaw/config.py

```
@@ -2,15 +2,40 @@
+    v = re.sub(r"^[vV]", "", v)
+    # Strip pre-release / build-metadata suffix (everything from the first
+    # hyphen or plus onward, per semver)
+    v = re.split(r"[-+]", v, maxsplit=1)[0]
+    parts = v.split(".")
+    try:
+        return tuple(int(p) for p in parts)
+    except ValueError:
+        raise ValueError(
+            f"Invalid version string '{original}': expected a numeric "
+            f"version like '1.0.0', got non-numeric component"
+        )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)